Framework/PythonInterface/plugins/algorithms/HB3AIntegrateDetectorPeaks.py

```diff
@@ -0,0 +1,203 @@
+                                 Constraints=constraints,
+                                 StartX=startX, EndX=endX,
+                                 EnableLogging=False)
+            except RuntimeError as e:
+                self.log().warning("Failed to fit workspace {}: {}".format(inWS, e))
+                continue
+
+            if fit_result.OutputStatus == 'success' and fit_result.OutputChi2overDoF < chisqmax:
+                __tmp_pw = CreatePeaksWorkspace(OutputType='LeanElasticPeak',
+                                                InstrumentWorkspace=inWS,
+                                                NumberOfPeaks=0,
+                                                EnableLogging=False)
+
+                _, A, x, s, _ = fit_result.OutputParameters.toDict()['Value']
+                _, errA, _, errs, _ = fit_result.OutputParameters.toDict()['Error']
+
+                if scan_log == 'omega':
+                    SetGoniometer(Workspace=__tmp_pw, Axis0=f'{x},0,1,0,-1', Axis1='chi,0,0,1,-1', Axis2='phi,0,1,0,-1',
+                                  EnableLogging=False)
+                else:
+                    SetGoniometer(Workspace=__tmp_pw, Axis0='omega,0,1,0,-1', Axis1='chi,0,0,1,-1', Axis2=f'{x},0,1,0,-1',
+                                  EnableLogging=False)
+
+                peak = __tmp_pw.createPeakHKL([run['h'].getStatistics().median,
+                                               run['k'].getStatistics().median,
+                                               run['l'].getStatistics().median])
+                peak.setWavelength(float(run['wavelength'].value))
+
+                integrated_intensity = A * s * np.sqrt(2*np.pi) * scale
+                peak.setIntensity(integrated_intensity)
+
+                # Convert correlation back into covariance
+                cor_As = (fit_result.OutputNormalisedCovarianceMatrix.cell(1,4)/100
+                          * fit_result.OutputParameters.cell(1,2) * fit_result.OutputParameters.cell(3,2))
+                # σ^2 = 2π (A^2 σ_s^2 + σ_A^2 s^2 + 2 A s σ_As)
+                integrated_intensity_error = np.sqrt(2*np.pi * (A**2 * errs**2 +  s**2 * errA**2 + 2*A*s*cor_As)) * scale
+                peak.setSigmaIntensity(integrated_intensity_error)
+
+                if integrated_intensity/integrated_intensity_error > signalNoiseMin:
+                    __tmp_pw.addPeak(peak)
+
+                    # correct q-vector using CentroidPeaksMD
+                    if optmize_q:
+                        __tmp_q_ws = HB3AAdjustSampleNorm(InputWorkspaces=inWS, NormaliseBy='None', EnableLogging=False)
+                        __tmp_pw = CentroidPeaksMD(__tmp_q_ws, __tmp_pw, EnableLogging=False)
+                        DeleteWorkspace(__tmp_q_ws, EnableLogging=False)
+
+                    if use_lorentz:
+                        peak = __tmp_pw.getPeak(0)
+                        lorentz = abs(np.sin(peak.getScattering() * np.cos(peak.getAzimuthal())))
```

Comment:
This is from the ILL Neutron Data Booklet section 2.9-10, part 4.1, equation 7.  But yes it's very different, there should actually be a factor of λ^3 in there, according to Buras, but since it's constant wavelength it doesn't matter.

The Lorentz Correction for constant wavelength should probably be added to `LorentzCorrection` since it's used in a few different places.